ironic/drivers/modules/drac/raid.py

```diff
@@ -879,7 +970,64 @@ def _check_node_raid_jobs(self, task):
         if not node.driver_internal_info['raid_config_job_ids']:
             if not node.driver_internal_info.get('raid_config_job_failure',
                                                  False):
-                self._resume_cleaning(task)
+                if 'raid_config_substep' in node.driver_internal_info:
+                    if node.driver_internal_info['raid_config_substep'] == \
+                            'delete_reset_config':
+                        controllers = list()
+                        jobs_required = False
+                        for controller_id in \
+                                node.driver_internal_info[
+                                    'raid_config_parameters']:
+                            controller_cap = clear_foreign_config(
+                                node, controller_id)
+                            controller = {'raid_controller': controller_id,
+                                          'is_reboot_required':
+                                              controller_cap[
+                                                  'is_reboot_required']}
+                            controllers.append(controller)
+                            jobs_required = jobs_required == controller_cap[
+                                'is_commit_required']
+
+                        if jobs_required:
```

Comment:
I think the above line needs to be:
if not jobs_required:
This will make it so that if there were no foreign drives, it will enter the code below, log the "No foreign drives" message and continue.

Reply:
Resolved this error